src/libdrakvuf/win-processes.c

```diff
@@ -313,23 +313,36 @@ static unicode_string_t* win_get_process_full_name(drakvuf_t drakvuf, addr_t epr
 
 char* win_get_process_name(drakvuf_t drakvuf, addr_t eprocess_base, bool fullpath)
 {
-    if ( fullpath )
-    {
-        unicode_string_t* fullname = win_get_process_full_name( drakvuf, eprocess_base );
+    unicode_string_t* fullname = win_get_process_full_name( drakvuf, eprocess_base );
 
-        if (fullname && fullname->contents && strlen((const char*)fullname->contents) > 0)
+    if (fullname && fullname->contents && strlen((const char*)fullname->contents) > 0)
+    {
+        // Replace 'proc_data->name' with 'fullname->contents'
+        // Moving ownership of fullname->contents to name for later cleanup
+        char* name = (char*)fullname->contents;
+        g_free( (gpointer)fullname );
```

Comment:
This shouldn't be a `g_free` but a `vmi_free_unicode_str` and only after you are done using `->contents`

Reply:
I've tried that and it leads to a Double free, [https://github.com/tklengyel/drakvuf/commit/84ad771b4c1857c34ffa7173faed5e23f93e781e](url)

It's confusing because this is almost identical flow to the one before , the only difference is the parent if condition is moved inside.

Reply:
This is identical to the original code flow of this function, I'm not sure why it fails, though I have tried what you pointed out, still results in memory leaks, and I tried to replace it with `vmi_free_unicode_str` in a previous commit , and it gave `Double free` error. [84ad771b4c1857c34ffa7173faed5e23f93e781e](https://github.com/tklengyel/drakvuf/commit/84ad771b4c1857c34ffa7173faed5e23f93e781e)